pkg/api/server/v1alpha2/results_test.go

```diff
@@ -42,14 +42,19 @@ func TestCreateResult(t *testing.T) {
 	req := &pb.CreateResultRequest{
 		Parent: "foo",
 		Result: &pb.Result{
-			Name: "foo/results/bar",
+			Name:        "foo/results/bar",
+			Annotations: map[string]string{"foo": "bar"},
 		},
 	}
 	t.Run("success", func(t *testing.T) {
 		got, err := srv.CreateResult(ctx, req)
 		if err != nil {
 			t.Fatalf("could not create result: %v", err)
 		}
+		got, err = srv.GetResult(ctx, &pb.GetResultRequest{Name: got.GetName()})
```

Comment:
Since this is likely going to be a common use case, let's also add a test filtering a result by the annotation data for completeness.